### PR TITLE
dind: ensure node certs are generated serially

### LINKED ISSUE #11274: networking test flake "certificate is valid for nettest-node-1, 172.17.0.3, not nettest-node-2"
Seen in https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check/6833:

`Oct  7 17:38:40.311: INFO: Error running &{/data/src/github.com/openshift/origin/_output/local/bin/linux/amd64/oc [oc logs --namespace=extended-test-get-flows-57ect-zuw9v --config=/tmp/extended-test-get-flows-57ect-zuw9v-user.kubeconfig flow-checklhsi6] []   Error from server: Get https://nettest-node-2:10250/containerLogs/extended-test-get-flows-57ect-zuw9v/flow-checklhsi6/flow-check: x509: certificate is valid for nettest-node-1, 172.17.0.3, not nettest-node-2`

This is the same symptom as #11025 but presumably not the same cause. This was in a post-#11061 tree, so it may actually be caused by a dind deployment bug this time...

@marun

### PR DESCRIPTION
**Pull request by marun**
The cert gen commands are not intended to be run concurrently.

Intended to fix #11274

cc: @liggitt @openshift/networking @stevekuznetsov

### CODE REVIEW

**Comment by marun** [comment]
[test]

**Comment by marun** [comment]
flakes #11094 #9490 #9548 

re-[test]

File: images/dind/node/openshift-generate-node-config.sh [hunk 1]
```diff
@@ -8,6 +8,14 @@ source /usr/local/bin/openshift-dind-lib.sh
 # Should set OPENSHIFT_NETWORK_PLUGIN
 source /data/network-plugin
 
+function lock() {
+  local config_path=$1
+
+  exec 200>"${config_path}"/.openshift-generate-node-config.exclusivelock
+
+  flock -n 200 && return 0 || return 1
```
Comment (stevekuznetsov) [inline images/dind/node/openshift-generate-node-config.sh]: What does masking the return code here achieve? The only difference is non-zero return codes get aliased as 1.

File: images/dind/node/openshift-generate-node-config.sh [hunk 2]
```diff
@@ -8,6 +8,14 @@ source /usr/local/bin/openshift-dind-lib.sh
 # Should set OPENSHIFT_NETWORK_PLUGIN
 source /data/network-plugin
 
+function lock() {
```
Comment (stevekuznetsov) [inline images/dind/node/openshift-generate-node-config.sh]: We probably want to namespace this function, I feel like we could have name collisions with `lock`

Comment (marun) [inline images/dind/node/openshift-generate-node-config.sh]: Done

File: images/dind/node/openshift-generate-node-config.sh (hunk 1 above)
Comment (marun) [inline images/dind/node/openshift-generate-node-config.sh]: Done

**Comment by marun** [comment]
flake #11240 

re-[test]

**Comment by marun** [comment]
@stevekuznetsov ptal

**Comment by stevekuznetsov** [comment]
LGTM

**Comment by danwinship** [comment]
Shouldn't we drop the lock at some point? AIUI, once we get the lock, this would hold it until the script exits

**Comment by marun** [comment]
@danwinship I'm not sure it's a problem to hold it until the file exits, but it can't hurt to unlock after config is generated.

**Comment by marun** [comment]
Closed in favor of inclusion in #11326.

**Pull request closed without merging**